criu/net.c

```diff
@@ -1818,6 +1819,52 @@ static inline int dump_iptables(struct cr_imgset *fds)
 	return 0;
 }
 
+static inline int dump_nftables(struct cr_imgset *fds)
+{
+	int ret = -1;
+	struct cr_img *img;
+	int img_fd;
+	FILE *fp;
+	struct nft_ctx *nft;
+
+	/* we not dump nftables if nft utility isn't present */
+	if (!check_cmd_exists("nft"))
+		return 0;
+
+	nft = nft_ctx_new(NFT_CTX_DEFAULT);
+	if (!nft)
+		return -1;
+
+	img = img_from_set(fds, CR_FD_NFTABLES);
+	img_fd = dup(img_raw_fd(img));
+	if (img_fd < 0) {
+		pr_perror("dup() failed");
+		goto nft_ctx_free_out;
+	}
```

Comment:
Why you need a dup here? Why you can't use the fd returned by img_raw_fd? You also forget to close img_fd.

Reply:
Ah ok I see in fdopen man, that fclose will close the fd. My bad.

Reply:
>  img_fd.

I've checked other places in CRIU, and realized that in all places we don't close descriptor that returned by img_raw_fd. Therefore, I need to dup descriptor -> then close dupped. I've used fclose() and fclose() always closes underlying fd.